File: app/src/main/java/com/hazem/wazaker/Fragments/AzkarFragment.java
package com.hazem.wazaker.Fragments;

import android.content.Intent;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.RecyclerView;

import com.hazem.wazaker.Adapters.AzkarAdapter;
import com.hazem.wazaker.activities.AzkarActivity;
import com.hazem.wazaker.Models.ListItem;
import com.hazem.wazkar.R;

import java.util.ArrayList;
import java.util.List;


public class AzkarFragment extends Fragment {


    private List<ListItem> items;


    public AzkarFragment() {
        // Required empty public constructor
    }


    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View v = inflater.inflate(R.layout.fragment_home, container, false);
        RecyclerView rec = v.findViewById(R.id.rvAzkar);
        AzkarAdapter azkarAdapter = new AzkarAdapter((ArrayList<ListItem>) items, getContext());
        rec.setAdapter(azkarAdapter);
        rec.setHasFixedSize(true);

        azkarAdapter.setListner(e -> {
            int index = (int) e;
            Intent zeker = new Intent(getActivity(), AzkarActivity.class);
            zeker.putExtra("index", index);
            startActivity(zeker);
        });

        return v;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        loadData();
    }

    private void loadData() {
        items = new ArrayList<>();
        items.add(new ListItem("أذكار الصباح "));
        items.add(new ListItem("أذكار المساء"));
        items.add(new ListItem("أذكار المسجد"));
        items.add(new ListItem("أذكار بعد الصلاه "));
        items.add(new ListItem("أذكار الأذان "));
        items.add(new ListItem("أذكار الوضوء"));
        items.add(new ListItem("الأدعيـة القرآنية"));
        items.add(new ListItem("الأدعيه النبويه "));
        items.add(new ListItem("أذكار الطعام"));
        items.add(new ListItem("أذكار المنزل"));
        items.add(new ListItem("أذكار النوم"));
        items.add(new ListItem("أذكار الإستيقاظ من النوم"));
        items.add(new ListItem("أذكار عظيمة"));
        items.add(new ListItem("دعاء ختم القرآن الكريم"));
        items.add(new ListItem("فضل الدعاء"));
        items.add(new ListItem("فضل تلاوه القرآن الكريم"));
        items.add(new ListItem("فضل الأذكار"));
        items.add(new ListItem("الرقيه الشرعيه"));
    }


}

File: app/src/main/java/com/hamza/Wazaker/ui/Fragments/AzkarFragment/AzkarFragment.kt
package com.hamza.Wazaker.ui.Fragments.AzkarFragment

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.LinearLayoutManager
import com.hamza.Wazaker.Data.Data_sets.AzkarListDataset
import com.hamza.Wazaker.ui.Adapters.AzkarListAdapter
import com.hamza.Wazaker.ui.listeners.ItemListner
import com.hamza.wazkar.databinding.FragmentAzkarBinding

class AzkarFragment : Fragment() {

    var _binding: FragmentAzkarBinding? = null
    val binding get() = _binding!!

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        // Inflate the layout for this fragment

        _binding = FragmentAzkarBinding.inflate(layoutInflater, container, false)

        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        setAdapter()
    }

    private fun setAdapter() {
        val layoutManager = LinearLayoutManager(context)
        binding.AzkarListRecycler.layoutManager = layoutManager
        val azkarListAdapter = AzkarListAdapter(AzkarListDataset.getAzkarList())
        binding.AzkarListRecycler.adapter = azkarListAdapter
        azkarListAdapter.setListner(object : ItemListner {
            override fun onItemCLicked(item: Any?) {
//                val index = e as Int
//                val zeker = Intent(context, ZekerActivity::class.java)
//                zeker.putExtra("index", index)
//                startActivity(zeker)
            }
        })
    }
}